baseplate/__init__.py

```diff
@@ -670,6 +670,11 @@ def make_child(
         """Return a child Span whose parent is this Span."""
         raise NotImplementedError
 
+    def with_tags(self, tags: Dict[str, Any]) -> "Span":
+        for k, v in tags.items():
+            self.set_tag(k, v)
+        return self
```

Comment:
Thanks for adding this 🙏  If it's not too much of a pain what do you think about pulling this change out into its own PR + adding a test (looks like others are in `tests/unit/core_tests.py`)?

Reply:
https://github.com/reddit/baseplate.py/pull/667